query_test.go

```diff
@@ -2986,8 +2984,7 @@ func TestGetDataBuffer(t *testing.T) {
 	require.NoError(t, err)
 	storedDataBuffer, ok := storedBuffer.([]int32)
 	require.True(t, ok)
-	require.Equal(t, ((*reflect.SliceHeader)(unsafe.Pointer(&dataBuffer)).Data),
-		((*reflect.SliceHeader)(unsafe.Pointer(&storedDataBuffer)).Data))
+	require.Equal(t, uintptr(slicePtr(dataBuffer)), uintptr(slicePtr(storedDataBuffer)))
```

Comment:
you should be able to remove these `uintptr` conversions as well, since both of them are now `unsafe.Pointer`s (I think?)

Reply:
Done